src/components/ProductItem/ProductItem.tsx

```diff
@@ -0,0 +1,128 @@
+import React from "react";
+import { useState, useEffect } from "react";
+
+import Button from "@components/Button";
+import { ButtonColor } from "@components/Button";
+import Card from "@components/Card";
+import axios from "axios";
+import { Link, useParams } from "react-router-dom";
+
+import styles from "./ProductItem.module.scss";
+
+interface IProduct {
+  id: number;
+  image: string;
+  title: string;
+  subtitle: string;
+  category: string;
+  content: string;
+}
+const ProductItem = () => {
+  const [allproducts, setAllProducts] = useState([]);
+  useEffect(() => {
+    const fetchAllproducts = async () => {
+      const result = await axios({
+        method: "get",
+        url: "https://fakestoreapi.com/products",
+      });
```

Comment:
эндпоинты обычно выносятся в отдельный файл конфига, чтобы к ним был упрощен доступ
файл `config/endpoints`
```ts
type EndpointType = {
  url: string;
  method: string;
};

const ENDPOINTS: Record<string, EndpointType> = {
  products: {
    method: 'GET',
    url: 'https://fakestoreapi.com/products',
  }
};
```
```suggestion
      const result = await axios(ENDPOINTS.products);
```